Fix layer filtering

jo-mueller opened the pull request:
Fixes #5 

Minor change that checks whether a selected layer in the viewer for fusion is actually an image or a label. I used a convenience function to retrieve the layers of the correct kind which is then called in both the load all and load selected functions:

```python
    def _get_compatible_layers(self, only_selected=False):
        if only_selected:
            return [l for l in self.viewer.layers.selection if isinstance(l, (Image, Labels))]
        else:
            return [l for l in self.viewer.layers if isinstance(l, (Image, Labels))]
```

I also added some warnings in case an invalid selection is passed. Let me know what you think :)

m-albert reviewed (changes requested):
Thanks for the PR 🙏

I like having a helper function as in the future there'll be other plugin widgets that'll need to do the same compatibility check.

Suggestion: Since loading always goes through the bottleneck of `load_layers`, I think the cleanest would be to do the filtering only there (currently there's some redundancy, see the comment in the code review).

jo-mueller commented:
Hi @m-albert ,

thanks for the review, that's indeed the better place to put it. I have moved the filtering there with the slight difference that I filter the input `layer` variable rather than `self.input_layers` as the former is used further down in the loading and the plugin seems to run into trouble when `layers` and `input_layers` get out of sync. 

I also removed the convenience function again since it is right now only used in this one place - one can re-introduce it if there is a need for it down the road :)

m-albert commented:
Looks great! Thanks a lot for this iteration.

> I have moved the filtering there with the slight difference that I filter the input layer variable rather than self.input_layers as the former is used further down in the loading and the plugin seems to run into trouble when layers and input_layers get out of sync.

Completely agreed that only filtered/confirmed layers should be included in `self.input_layers`. Indeed there are callbacks on added layers monitoring for changes in transformations after adding them to the widget.

> I also removed the convenience function again since it is right now only used in this one place - one can re-introduce it if there is a need for it down the road :)

Sounds good.

Cool, merging this now. Thanks a lot for contributing!
